CMakeLists.txt

```diff
@@ -0,0 +1,459 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+cmake_minimum_required(VERSION 3.22.1 FATAL_ERROR)
+
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.13)
+  cmake_policy(SET CMP0077 NEW)
+  set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+endif()
+
+##############################################################################
+# - Download and initialize RAPIDS CMake helpers -----------------------------
+
+file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-22.08/RAPIDS.cmake
+    ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+include(${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+include(rapids-cmake)
+include(rapids-cpm)
+include(rapids-cuda)
+include(rapids-export)
+include(rapids-find)
+
+##############################################################################
+# - User Options  ------------------------------------------------------------
+
+option(BUILD_SHARED_LIBS "Build cuNumeric shared libraries" ON)
+option(CUNUMERIC_EXCLUDE_TBLIS_FROM_ALL "Exclude tblis targets from cuNumeric's 'all' target" OFF)
+option(CUNUMERIC_EXCLUDE_OPENBLAS_FROM_ALL "Exclude OpenBLAS targets from cuNumeric's 'all' target" OFF)
+option(CUNUMERIC_EXCLUDE_LEGATE_CORE_FROM_ALL "Exclude legate.core targets from cuNumeric's 'all' target" OFF)
+
+##############################################################################
+# - Project definition -------------------------------------------------------
+
+project(CUNUMERIC VERSION 22.07.00 LANGUAGES C CXX)
+
+# Write the version header
+rapids_cmake_write_version_file(include/cunumeric/version_config.hpp)
+
+##############################################################################
+# - Build Type ---------------------------------------------------------------
+
+# Set a default build type if none was specified
+rapids_cmake_build_type(Release)
+
+# Needed to integrate with LLVM/clang tooling
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# ############################################################################
+# * conda environment --------------------------------------------------------
+rapids_cmake_support_conda_env(conda_env MODIFY_PREFIX_PATH)
+
+##############################################################################
+# - Dependencies -------------------------------------------------------------
+
+# add third party dependencies using CPM
+rapids_cpm_init()
+
+find_package(OpenMP)
+
+option(Legion_USE_CUDA "Use CUDA" ON)
```

Comment:
I'm of two minds on the naming of these variables. Generally, I prefer to name things locally as 'Cunumeric_USE_CUDA', etc. AFAICT, Legion_USE_CUDA is never imported as a variable so we are free to call it whatever we want.  Someone configuring cunumeric shouldn't have to reference flags named after a non-public dependency.  I lean towards naming these variables as Cunumeric_XYZ, but I don't have strong feelings on it.